src/Symfony/Component/Validator/Constraints/VideoValidator.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Process\Process;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @author Kev <https://github.com/symfonyaml>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class VideoValidator extends FileValidator
+{
+    public function validate(mixed $value, Constraint $constraint): void
+    {
+        if (!$constraint instanceof Video) {
+            throw new UnexpectedTypeException($constraint, Video::class);
+        }
+
+        $violations = \count($this->context->getViolations());
+
+        parent::validate($value, $constraint);
+
+        $failed = \count($this->context->getViolations()) !== $violations;
+
+        if ($failed || null === $value || '' === $value) {
+            return;
+        }
+
+        if (null === $constraint->minWidth && null === $constraint->maxWidth
+            && null === $constraint->minHeight && null === $constraint->maxHeight
+            && null === $constraint->minPixels && null === $constraint->maxPixels
+            && null === $constraint->minRatio && null === $constraint->maxRatio
+            && $constraint->allowSquare && $constraint->allowLandscape && $constraint->allowPortrait
+        ) {
+            return;
+        }
+
+        $process = new Process([
+            'ffprobe',
+            '-v', 'error',
+            '-select_streams', 'v',
+            '-show_entries', 'stream=index,codec_type,codec_name,width,height',
+            '-show_entries', 'format=format_name',
+            '-of', 'json',
+            (string) $value,
+        ]);
+        $process->run();
+
+        if (!$process->isSuccessful()) {
+            $this->context->buildViolation($constraint->corruptedMessage)
+                ->setCode(Video::CORRUPTED_VIDEO_ERROR)
+                ->addViolation();
+
+            return;
+        }
+
+        $meta = json_decode($process->getOutput(), true) ?: [];
+        $streams = $meta['streams'] ?? [];
+        $formats = explode(',', strtolower($meta['format']['format_name'] ?? 'unknown'));
+
+        if (!($streams[0]['width'] ?? false) || !($streams[0]['height'] ?? false)) {
```

Comment:
Huge error here, $streams[0] is allowed to not be the default video stream, you must check the default flag at $stream['disposition']['default'] or calcule it yourself if its not present.

Reply:
If you read the code below, you'll see that we reject videos that have multiple streams.
This means we don't have to care about the default stream.
No bug here.

I appreciate that you care about us providing a video validator that makes sense. But I don't think the rage that your messages express is deserved. Both on your side - it must be stressful to have so much energy to yell at others - and for the others, which don't have to be the recipient of your anger.

Please keep a respectful tone.

Reply:
I hope you understand that excluding multiple video streams videos validation makes this validator is even less usable, unless, of course, this is your personal project. But somehow I hope its not and that code aims to be usable by other devs, in that case, it would be nice to see this validator limitations mentioned in the docs and/or blog.

In regards for the second paragraph I can't comment much about narcissists hurt feelings. Maybe be less sensitive?